src/components/Dish/Dish.js

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+
+import dataApi from 'api/data';
+import DishCard from './DishCard';
+import Loader from 'components/common/Loader';
+
+import style from './Dish.module.css';
+
+class Dish extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      dishesRaw: [],
+      dishes: [],
+      prevSearchDishValue: '',
+      isLoading: false,
+      requestErrMsg: null,
+    };
+  }
+
+  componentDidMount() {
+    this._loadInitDishes();
+  }
+
+  componentDidUpdate() {
+    if (this.state.prevSearchDishValue !== this.props.searchDishValue) {
+      this.filterDishesBySearchParam(this.props.searchDishValue);
+    }
+  }
+
+  _loadInitDishes = async () => {
```

Comment:
I do not recommend using underscores in methos naming because underscore indicates that the method is private which is really not. 
Also, if you are interested in more details you can check it here: 
https://github.com/airbnb/javascript/issues/1024
https://github.com/airbnb/javascript/issues/490